src/SSMSPlusSearch/UI/SchemaSearchControlVM.cs

```diff
@@ -167,16 +168,57 @@ private async Task ExecuteSearchAsync()
                 await Task.Run(() =>
                  {
                      var filter = new FilterContext(Filter, ComboMatchVM.GetSelectedValues(), ComboObjectsVM.GetSelectedValues(), SchemaObjectsVM.GetSelectedValues());
-                     SearchResultsVM = FilterResultService.Filter(_allDdResults, filter).Select(p => new SearchFilterResultVM(p, filter)).ToList();
+                     var result = FilterResultService.Filter(_allDdResults, filter).Select(p => new SearchFilterResultVM(p, filter));
+
+                     //Sorting 
+                     Func<SearchFilterResultVM, dynamic> sortExpression = null;
+                     switch (CurrentSortColumn)
+                     {
+                         case ("Name"):
+                             sortExpression = (x => x.SearchResult.Name);
+                             break;
+
+                         case ("SchemaName"):
+                             sortExpression = (x => x.SearchResult.SchemaName);
+                             break;
+
+                         case ("DisplayType"):
+                             sortExpression = (x => x.SearchResult.DisplayType);
+                             break;
+
+                         case ("ModificationDateStr"):
+                             sortExpression = (x => x.SearchResult.ModificationDate);
+                             break;
+
+                         case ("RichSmallDefinition"):
+                             sortExpression = (x => x.SearchResult.RichSmallDefinition.AsString);
+                             break;
+
+                         default:
+                             sortExpression = (x => x.SearchResult.Name);
+                             break;
+                     }
+
+                     // sort direction
+                     if (CurrentSortDirection == ListSortDirection.Descending)
+                     {
+                         SearchResultsVM = result.OrderByDescending(sortExpression).ToList();
+                     }
+                     else
+                     {
+                         SearchResultsVM = result.OrderBy(sortExpression).ToList();
+                     }
+

```

Comment:
I think we can handle the "keep sorting" without adding this much code in the view model